metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorPartitionService.java

```diff
@@ -268,7 +268,7 @@ public PartitionsSaveResponse savePartitions(
             // Existing partition map
             Map<String, Partition> existingPartitionMap = Collections.emptyMap();
 
-            if (partitionsSaveRequest.getCheckIfExists()) {
+            if (partitionsSaveRequest.getCheckIfExists() || partitionsSaveRequest.getAlterIfExists()) {
```

Comment:
I'm going to channel @mprimi here... These changes should at least be in a different commit as you're changing behavior (not just code formatting) and so if we ever go to search the git history this will be totally lost as we think this commit will only be related to the events. You should probably do a `git reset --soft HEAD~1` on your local branch and then commit each of the changes as separate commits with different messages. Then instead of squash merging we can rebase and fast-forward merge and keep the commits separate.